pkg/apis/core/v1beta1/types.go

```diff
@@ -256,6 +256,58 @@ const (
 // DataVolumeCloneSourceSubresource is the subresource checked for permission to clone
 const DataVolumeCloneSourceSubresource = "source"
 
+//StorageProfile provides a CDI specific recommendation for storage parameters
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:resource:scope=Cluster
+type StorageProfile struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StorageProfileSpec   `json:"spec"`
+	Status StorageProfileStatus `json:"status,omitempty"`
+}
+
+//StorageProfileSpec defines specification for StorageProfile
+type StorageProfileSpec struct {
+	// ClaimPropertySets is a provided set of properties applicable to PVC
+	ClaimPropertySets []ClaimPropertySet `json:"claimPropertySets,omitempty"`
+}
+
+//StorageProfileStatus provides the most recently observed status of the StorageProfile
+type StorageProfileStatus struct {
+	// The StorageClass name for which capabilities are defined
+	StorageClass *string `json:"storageClass,omitempty"`
+	// The Storage class provisioner plugin name
+	Provisioner *string `json:"provisioner,omitempty"`
+	// ClaimPropertySets computed from the spec and detected in the system
+	ClaimPropertySets []ClaimPropertySet `json:"claimPropertySets,omitempty"`
+}
+
+// ClaimPropertySet is a set of properties applicable to PVC
+type ClaimPropertySet struct {
+	// AccessModes contains the desired access modes the volume should have.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
+	// +optional
+	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes,omitempty" protobuf:"bytes,1,rep,name=accessModes,casttype=PersistentVolumeAccessMode"`
+	// volumeMode defines what type of volume is required by the claim.
+	// Value of Filesystem is implied when not included in claim spec.
+	// +optional
+	VolumeMode *corev1.PersistentVolumeMode `json:"volumeMode,omitempty" protobuf:"bytes,6,opt,name=volumeMode,casttype=PersistentVolumeMode"`
```

Comment:
Should snapshot support be included?  If not part of the spec, should it at least be supported in the status?

Reply:
What kind of snapshot support? Right now the main goal is to support recommended parameters that can be used to create a DV/PVC. So to create a DV without specifying accessModes/volumeMode. 

Current scope does not cover snapshot. And as I said before - we can always extend in the future (I think adding things to this Resource is backwards compatible). 

If you think it will be better to extend the scope of the change, then we need to have some discussion.

Reply:
Perhaps it is just an implementation detail but I think all other things being equal we should prefer a storageclass that supports snapshots over ones that don't.

And I think it may be nice to report snapshot support in the status

Reply:
We can add the snapshot support at a later time. I do agree it will be nice to prefer storage with snapshot support in the future.